src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1406,4 +1411,48 @@ public static void modifyAttributes()
             FORGE_ATTRIBUTES.put(k, newBuilder.build());
         });
     }
+    
+    /**
+     * Called when dimension jsons are parsed.
+     * Creates a copy of the worldgen settings and its dimension registry,
+     * where dimensions that specify that they should use the server seed will use the server seed
+     * instead of the seed that mojang requires be specified in the json.
+     **/
+    public static WorldGenSettings loadDimensionsWithServerSeed(WorldGenSettings wgs)
+    {
+        // get the original worldgen settings' settings
+        long seed = wgs.seed();
+        boolean generateFeatures = wgs.generateFeatures();
+        boolean generateBonusChest = wgs.generateBonusChest();
+        MappedRegistry<LevelStem> originalRegistry = wgs.dimensions();
+        Optional<String> legacyCustomOptions = wgs.legacyCustomOptions;
+        
+        // make a copy of the dimension registry; for dimensions that specify that they should use the server seed instead
+        // of the hardcoded json seed, recreate them with the correct seed
+        MappedRegistry<LevelStem> seededRegistry = new MappedRegistry<>(Registry.LEVEL_STEM_REGISTRY, Lifecycle.experimental());
+        for (Entry<ResourceKey<LevelStem>, LevelStem> entry : originalRegistry.entrySet())
+        {
+            ResourceKey<LevelStem> key = entry.getKey();
+            LevelStem dimension = entry.getValue();
+            if (dimension.useServerSeed)
+            {
+                seededRegistry.register(key, new LevelStem(dimension.typeSupplier(), dimension.generator().withSeed(seed)), originalRegistry.lifecycle(entry.getValue()));
```

Comment:
Shouldn't this continue to pass in the parameter into `LevelStem` that it is using the server seed?

Reply:
Good catch, added this in latest push